protocol/messenger_handler.go

```diff
@@ -3376,6 +3379,18 @@ func (m *Messenger) handleSyncWatchOnlyAccount(message *protobuf.SyncAccount, fr
 		return nil, err
 	}
 
+	if m.config.accountsFeed != nil {
+		var eventType accountsevent.EventType
+		if acc.Removed {
```

Comment:
I have absolutely no idea how this sync thingy ends up actually adding or removing the account, are you sure we won't get 2 notifications (here and wherever we had them before) for a single addition/removal?

Reply:
I actually was under the impression we had something like a "removeLater" flag when removing through sync, so the actual removal got done on the next app start, but I'm not super sure about that...

Reply:
No, that's not the case from what I experienced so far (from UI and checking the go code). When one of the synced devices adds/removes any accounts, it's immediately done, updates the DB and sends a signal to the UI of the same (that's why this function returns the acc).